dxp/resource_serving.md

```diff
@@ -0,0 +1,161 @@
+This filter was used to serve _JavaScript bundles_ (sets of `.js` files that are bundled into a single resource) which were identified in the `portal.properties` file with the `javascript.bundle.ids` property. Out of the box, DXP came configured with two bundles: `javascript.barebone.files` and `javascript.everything.files`.
+
+Each bundle had a location in the server's URL space defined by the portal property `javascript.bundle.dir`. So, for example, both `javascript.bundle.dir[javascript.barebone.files]` and `javascript.bundle.dir[javascript.everything.files]` were located in the root of the server (`/`). This is the place from which the files in the bundle are retrieved when requested.
+
+To finish with, any `.js` resource you request (no matter what the URL is) which has valid `minifierType` and `minifierBundleId`/`bundleId` request parameters returns a cached copy of the JavaScript bundle aggregate. Any value is valid for `minifierType` while only defined bundles can be passed as `minifierBundleId`/`bundleId`.
+
+However, as it is now, DXP doesn't define any file in `portal.properties` for any of the two predefined bundles because they were migrated to a `DynamicInclude` service registration. Also, the mechanism is deprecated, though it still works if someone configures it.
+
+Note that the caches of this filter are invalidated by means of the [PortalWebResourcesUtil.getLastModified("js")](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/servlet/PortalWebResourcesUtil.java#L40) method, which looks up all registered [PortalWebResources](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/servlet/PortalWebResources.java) services with the `"js"` `resourceType` to find the latest modified one and the time when it was updated.
+
+#### JavaScript Single File
+
+When a single `.js` file is requested and the `minifierType` is set to `js`, the filter will minify the JavaScript file contents delegating to the [MinifierUtil.minifyJavascript() method](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/minifier/MinifierUtil.java#L45), which invokes the configured minifier (the OSGi service implementing the [JavaScriptMinifier](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/minifier/JavaScriptMinifier.java) interface with highest service ranking).
+
+Currently there are two implementations of `JavaScriptMinifier` which come out of the box with DXP: [YahooJavaScriptMinifier](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-minifier/src/main/java/com/frontend/js/minifier/YahooJavaScriptMinifier.java) and [GoogleJavaScriptMinifier](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-minifier/src/main/java/com/frontend/js/minifier/GoogleJavaScriptMinifier.java), being the last one (Google) the default one.
```

Comment:
"being the last one (Google) the default one" → "with the last one (Google) being the default"

Or you could just avoid the construction entirely:

> Currently there are two implementations of `JavaScriptMinifier` which come out of the box with DXP: [`YahooJavaScriptMinifier`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-minifier/src/main/java/com/frontend/js/minifier/YahooJavaScriptMinifier.java) and [`GoogleJavaScriptMinifier`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-minifier/src/main/java/com/frontend/js/minifier/GoogleJavaScriptMinifier.java) (the default).

Reply:
Funny. I used such a complex construction because when editing in markdown, the term `GoogleJavaScriptMinifier` is very far from `(the default)` (because there's the huge link between them) so my :brain:  lost track of it and though I needed to specify more :sweat_smile: .